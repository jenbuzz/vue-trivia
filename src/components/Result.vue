<template>
    <div class="text-center">
        <p>
            You got <strong>{{ correctAnswers }}</strong> correct!
        </p>
        <p>Thank you for playing!</p>
        <b-button @click="reset" variant="info">
            Play again!
        </b-button>
    </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';

@Component
export default class Result extends Vue {
    @Prop() private readonly correctAnswers!: number;
    @Prop() private reset!: Function;
}
</script>
