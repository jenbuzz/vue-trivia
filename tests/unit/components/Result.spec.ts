import { createLocalVue, shallowMount } from '@vue/test-utils';
import BootstrapVue from 'bootstrap-vue';
import Result from '@/components/Result.vue';

const localVue = createLocalVue();
localVue.use(BootstrapVue);

describe('Component: Result', () => {
    it('renders', () => {
        const wrapper = shallowMount(Result, {
            localVue,
            propsData: { correctAnswers: 2, reset: jest.fn() },
        });

        expect(wrapper).toMatchSnapshot();
    });
});
