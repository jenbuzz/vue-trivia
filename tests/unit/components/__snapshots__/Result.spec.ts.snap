// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Result renders 1`] = `
<div class="text-center">
  <p>
    You got <strong>2</strong> correct!
  </p>
  <p>Thank you for playing!</p>
  <b-button-stub variant="info" type="button" tag="button" target="_self">
    Play again!
  </b-button-stub>
</div>
`;
